src/skeptic/lib.rs

```diff
@@ -204,6 +218,12 @@ fn extract_tests_from_file(path: &Path) -> Result<DocTest, IoError> {
             }
             Event::Text(text) => {
                 if let Buffer::Code(ref mut buf) = buffer {
+                    if !text.contains("main()") {
```

Comment:
this will break apart if given codeblock does not contain `main()` which is quite often if one uses `quick_main` or any other macro that implements a main. 

All in all I would suggest counting from the beginning of code block start (this is quite easily done by observing the parser state machine).

Reply:
Hi budziq, can I get a clarification on the requirements here please?
Given the following file as input:
`
    let lines = &[
        "Rust code that should panic when running it.",//1
        "",
        "```rust,should_panic",
        "fn main() {",//4
        "    panic!(\"I should panic\");",
        "}",
        "```",
        "",
        "Rust code that should panic when compiling it.",//9
        "",
        "```rust,no_run,should_panic",
        "fn add(a: u32, b: u32) -> u32 {",
        "    a + b",
        "}",
        "",
        "fn main() {",//16
        "    add(1);",
        "}",
        "```",
    ];
`
are lines 4 & 16 the desired start, or is it 1 & 9? Or Neither?

Reply:
Hi @Borginator,

We would lik to have test number pointing to the start of the given codeblock so in this case it would be lines 3 and 11.